projects/smoke_ltc6811/src/main.c

```diff
@@ -0,0 +1,200 @@
+      LOG_DEBUG("CELL#%zu DELTA %d (MIN=%d, MAX=%d)\n", cell, delta, s_sample_bounds[cell].min,
+                s_sample_bounds[cell].max);
+    }
+  }
+
+  s_num_samples++;
+  if (s_num_samples == max_samples) {
+    prv_reset_sample_bounds();
+    s_num_samples = 0;
+  }
+  return STATUS_CODE_OK;
+}
+
+static void prv_dump_voltages(uint16_t *result_arr, size_t len, void *context) {
+  if (!status_ok(prv_extract_and_dump_readings(result_arr, len, SMOKE_LTC_AFE_NUM_VOLTAGE_SAMPLES)))
+    return;
+  if (s_num_samples != 0) {
+    if (!status_ok(ltc_afe_request_cell_conversion(&s_afe))) return;
+  } else {
+    LOG_DEBUG("==END OF VOLTAGE SAMPLES==");
+    LOG_DEBUG("==START OF TEMPERATURE SAMPLES==");
+    if (!status_ok(ltc_afe_request_aux_conversion(&s_afe))) return;
+  }
+}
+
+static void prv_dump_temps(uint16_t *result_arr, size_t len, void *context) {
+  if (!status_ok(prv_extract_and_dump_readings(result_arr, len, SMOKE_LTC_AFE_NUM_TEMP_SAMPLES)))
+    return;
+  if (s_num_samples != 0) {
+    if (!status_ok(ltc_afe_request_aux_conversion(&s_afe))) return;
+  } else {
+    LOG_DEBUG("==END OF TEMPERATURE SAMPLES==");
+    // TODO(SOFT-222): Could test cell discharge here as well ...
+  }
+}
+
+static StatusCode prv_ltc_init(void) {
+  status_ok_or_return(gpio_init());
+  interrupt_init();
+  soft_timer_init();
+  event_queue_init();
+
+  LtcAfeSettings afe_settings = {
+    .mosi = SMOKE_LTC_AFE_SPI_MOSI,
+    .miso = SMOKE_LTC_AFE_SPI_MISO,
+    .sclk = SMOKE_LTC_AFE_SPI_SCLK,
+    .cs = SMOKE_LTC_AFE_SPI_CS,
+
+    .spi_port = SMOKE_LTC_AFE_SPI_PORT,
+    .spi_baudrate = SMOKE_LTC_AFE_SPI_BAUDRATE,
+    .adc_mode = SMOKE_LTC_AFE_ADC_MODE,
+
+    // Because these need to be the max size, should do initialize this way
+    .cell_bitset = { 0 },
+    .aux_bitset = { 0 },
+
+    .num_devices = SMOKE_LTC_AFE_NUM_DEVICES,
+    .num_cells = SMOKE_LTC_AFE_NUM_CELLS,
+
+    .ltc_events = { .trigger_cell_conv_event = SMOKE_LTC_AFE_TRIGGER_CELL_CONV_EVENT,
+                    .cell_conv_complete_event = SMOKE_LTC_AFE_CELL_CONV_COMPLETE_EVENT,
+                    .trigger_aux_conv_event = SMOKE_LTC_AFE_TRIGGER_AUX_CONV_EVENT,
+                    .aux_conv_complete_event = SMOKE_LTC_AFE_AUX_CONV_COMPLETE_EVENT,
+                    .callback_run_event = SMOKE_LTC_AFE_CALLBACK_RUN_EVENT,
+                    .fault_event = SMOKE_LTC_AFE_FAULT_EVENT },
+
+    .cell_result_cb = prv_dump_voltages,
+    .aux_result_cb = prv_dump_temps,
+    .result_context = NULL,
+  };
+
+  for (int i = 0; i < SMOKE_LTC_AFE_NUM_DEVICES; i++) {
+    afe_settings.cell_bitset[i] = SMOKE_LTC_AFE_INPUT_BITSET_FULL;
+    afe_settings.aux_bitset[i] = SMOKE_LTC_AFE_INPUT_BITSET_FULL;
+  }
+
+  status_ok_or_return(ltc_afe_init(&s_afe, &afe_settings));
+
+  return STATUS_CODE_OK;
+}
+
+static void prv_log_event(const Event *event) {
+  if (SMOKE_LTC_SHOULD_LOG_EVENTS && event->id < NUM_SMOKE_LTC_EVENTS) {
```

Comment:
would be nice to have options to log only certain events :)

Reply:
I could have it such that an empty string in the mapping => no print for that event. Not sure if that's a good idea tho

Reply:
Your change seems fine